### PR TITLE
Fix part of #1587: Made some intent/bundle/saved/tags instance keys consistent.

### PR DESCRIPTION
**Pull request by ayush0402**
## Explanation

Made some keys consistent in various Activities and Fragments.
Fixes part of #1587 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by ayush0402** [comment]
@rt4914 , PTAL

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="false" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
```
Comment (prayutsu) [inline .idea/misc.xml]: Revert these changes.

File: .idea/runConfigurations.xml [hunk 2]
```diff
@@ -3,10 +3,11 @@
   <component name="RunConfigurationProducerService">
     <option name="ignoredProducers">
       <set>
+        <option value="com.android.tools.idea.compose.preview.runconfiguration.ComposePreviewRunConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.AllInPackageGradleConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestClassGradleConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestMethodGradleConfigurationProducer" />
```
Comment (prayutsu) [inline .idea/runConfigurations.xml]: Revert these too.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@ayush0402 Left some comments.
Also please link the issue with this PR

**Comment by prayutsu** [comment]
@ayush0402 To fix the failing unit tests, identify the failing test file and then try to find the cause of the error. To run the tests locally, read our [wiki instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing).

**Comment by rt4914** [comment]
@prayutsu Assign it to me or Akshay once approved from your side.

**Comment by prayutsu** [comment]
@ayush0402 You should not resolve conversation threads by yourself, only the person who started the thread should it, it helps in keeping track of the changes they requested. And once you make the changes as requested by the reviewer, you should reply with "Done" or "fixed" so that the reviewer is aware that you have made changes. 
Also, if you want any reviewers to take a look, then please drop a PTAL comment and tag the reviewer, so that oppiabot can assign the PRs to them.

**Comment by ayush0402** [comment]
I will keep that in mind for next time. I made the requested changes. @prayutsu PTAL.

File: .idea/misc.xml [hunk 3]
```diff
@@ -6,4 +6,4 @@
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (prayutsu) [inline .idea/misc.xml]: Revert these too.

File: .idea/runConfigurations.xml [hunk 4]
```diff
@@ -9,4 +9,4 @@
       </set>
     </option>
   </component>
-</project>
+</project>
```
Comment (prayutsu) [inline .idea/runConfigurations.xml]: Ditto.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@ayush0402 PTAL. It looks like you missed adding the extra blank line while undoing the changes.
All files must end with an empty new line (also called EOF).
Also please note that the issue hasn't been linked yet. You need to correct the keyword in the PR explanation -- it should be `Fixes part of #issueNumber` instead of `Fix part of #issueNumber`. 
Thanks!

**Comment by ayush0402** [comment]
@prayutsu PTAL. Made requested changes.

**Review by prayutsu — APPROVED** [review]
LGTM, thanks @ayush0402

**Comment by prayutsu** [comment]
@ayush0402 I have resolved all the conversations as of now, but you should always reply to the conversation threads with "Done" or "Fixed" when you commit the suggested changes so that the reviewer can resolve the conversation.

**Review by fsharpasharp — APPROVED** [review]
Approving for the changes in IntentFactoryShimImpl.kt

**Comment by anandwana001** [comment]
Please follow the first point in the PR checklist and mark it did once done. 
Assign me back once Rajat reviews the PR.

**Comment by ayush0402** [comment]
@anandwana001 Done. PTAL.

**Review by rt4914 — APPROVED** [review]
@ayush0402 Nice work. Thanks.

**Review by anandwana001 — APPROVED** [review]
LGTM

**Pull request merged**